fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/sorting/RecordQueryPlannerSortConfiguration.java

```diff
@@ -49,11 +59,15 @@ protected RecordQueryPlannerSortConfiguration() {
      * This method returns {@code true}.
      *
      * A subclass can override this method to use more complex criteria to decide.
-     * @param query the query that cannot be planned without a sort
      * @return whether to allow non-index sorting
      */
-    public boolean shouldAllowNonIndexSort(@Nonnull RecordQuery query) {
-        return true;
+    public boolean shouldAllowNonIndexSort() {
```

Comment:
I think the intention here was to allow for subclasses with more complex decision trees (based on the query). This becomes hard to serialize, so I think we want to instead encourage users to look at their query, and then based on that, decide whether to allow non-index sorts or not. I couldn't really find a great way of serializing this information otherwise.